mobify/progressive-web-scaffold
# Add a flag to opt you into auto fetching the HTML

**JIRA**: N/A

## Changes
- Add a `autoFetchHTMLFromOriginPage` flag to opt you into auto fetching the HTML, instead of opting you in by default

## How to test-drive this PR
- Run the scaffold, make sure the page is still fetched.
- Change `autoFetchHTMLFromOriginPage` from `true` to `false`, and make sure the page _isn't_ fetched.

## Background info

When I initially jumped into the scaffold when writing some documentation, it took me a very long time to understand where the actual page was being fetched for each of our routes. I eventually realized that this was happening "automagically" in app-provider.

To me, this goes against something in our developer values, ["explicit is better than implicit"](https://github.com/mobify/developer-values#readability-counts).

To me, this should be self documenting, hence why I've made the flag name very explicit, and also made it so that you have to set it on each route - it will force developers to take notice when adding new routes.

Personally, I don't believe this PR goes far enough - I think people should be able to go into their container and immediately understand where the data is coming from. At least with this change, people will see it pretty quickly when heading to the router.

If we want to provide easier mechanisms to auto-fetch the HTML, I think we could do that in a way that makes it much more clear in the containers. The question is, should we even bother with this intermediate step, or should we go straight to the containers?
